src/main/RemoteRequest.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * RemoteRequest is used to for http requests on a remote server which can be
+ * fetched in chunks, e.g. using a Range request.
+ * @flow
+ */
+'use strict';
+
+import Q from 'q';
+
+type Chunk = {
+  start: number;
+  stop: number;
+  buffer: Object; // TODO: more efficient data store
```

Comment:
The main difference is that RemoteFile is file oriented, and much of its functionality is focused around parsing and storing bytes. RemoteRequest deals with json, not byte data. Do you think these differences can be reconciled as one class?